server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -231,6 +233,14 @@ final class DefaultSearchContext extends SearchContext {
         queryBoost = request.indexBoost();
         this.lowLevelCancellation = lowLevelCancellation;
         this.requestToAggReduceContextBuilder = requestToAggReduceContextBuilder;
+        if (indexService.getIndexSettings() != null && clusterService != null) {
```

Comment:
AFAIK `indexService.getIndexSettings()` should never return `null`, have you run into that?